net-vpn/i2p/i2p-2.6.0.ebuild

```diff
@@ -0,0 +1,252 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="test"
+
+inherit java-pkg-2 systemd toolchain-funcs
+
+DESCRIPTION="A privacy-centric, anonymous network"
+HOMEPAGE="https://geti2p.net"
+SRC_URI="https://files.i2p-projekt.de/${PV}/i2psource_${PV}.tar.bz2"
+
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+LANGS=(
+	ar az bg ca cs da de el en es es-AR et fa fi fr gl he hi hr hu id it ja ko ku mg nb nl nn pl pt pt-BR ro ru sk sl sq
+	sr sv tk tr uk vi zh zh-TW
+)
+IUSE="${LANGS[@]/#/l10n_}"
+
+CP_DEPEND="
+	dev-java/bcprov:0
+	dev-java/hashcash:1
+	dev-java/httpcomponents-client:4
+	dev-java/identicon:1
+	dev-java/java-getopt:1
+	dev-java/java-service-wrapper:0
+	dev-java/jbcrypt:0
+	dev-java/json-simple:2.3
+	dev-java/json-smart:1
+	dev-java/jsonrpc2-base:1
+	dev-java/jsonrpc2-server:1
+	dev-java/jstl:0
+	dev-java/jstl-api:0
+	dev-java/minidns-core:1
+	dev-java/zxing-core:3
+	dev-java/zxing-javase:3
+	sys-devel/gettext:0[java]
+	www-servers/tomcat:9
+"
+# jdk-11 for bug #932030
+DEPEND="
+	dev-libs/gmp:0=
+	${CP_DEPEND}
+	>=virtual/jdk-11:*
+	test? (
+		dev-java/hamcrest:0
+		dev-java/junit:4
+		dev-java/mockito:4
+	)
+"
+BDEPEND="
+	>=dev-java/ant-1.10.14-r3:0
+	test? (
+		>=dev-java/ant-1.10.14-r3:0[junit4]
+	)
+"
+RDEPEND="
+	${CP_DEPEND}
+	acct-user/i2p
+	acct-group/i2p
+	>=virtual/jre-11:*
+"
+
+PATCHES=( "${FILESDIR}/force-gentoo-classpath.patch" )
+
+DOCS=( README.md history.txt )
+
+src_prepare() {
+	default # apply PATCHES
+	java-pkg-2_src_prepare
+
+	# remove most bundled, excepted the next ones.
+	# apps/addressbook/java/src/net/metanotion too much code drift
+	# apps/i2psnark/java/src/org/klomp/snark too much code drift
+	# apps/jrobin need rrd4j ebuild
+	# apps/routerconsole/java/src/{com,edu} too much code drift
+	# {core,router}/java/src/com/southernstorm/noise use internal symbols
+	# core/java/src/freenet too much code drift
+	# core/java/src/gnu/crypto too much code drift
+	# router/java/src/com/maxmind changed interface
+	# router/java/src/org/cybergarage unable to find version 3
+	# router/java/src/org/freenetproject too big to pull
+	# router/java/src/org/xlattice changed interface
+	java-pkg_clean ! \
+		-path "./apps/jetty/jetty-distribution-*" # need to package jetty
+	( cat >> override.properties || die 'set unbundled properties' ) <<- EOF
+		require.gettext=true
+		with-libgetopt-java=true
+		with-libjakarta-taglibs-standard-java=true
+		with-libjson-simple-java=true
+		with-libtomcat9-java=true
+		with-gettext-base=true
+		# with-geoip-database=true need std geoip use
+		# with-libjetty9-java=true needs a jetty ebuild
```

Comment:
doesn't it also need [ant.build.javac.source and ant.build.javac.target](https://github.com/gentoo/gentoo/pull/37945/files#diff-b94ac2e4fdd4bb3cb637d47df5404de960aa7e05dd54e998a5c42376f6ef707fR62-R63)?
to get this working, any eventually hardcoded `-source` and `-target` attributes must be removed from the `javac` elements in `build.xml`.
this was another part of `BSFIX` provided by `java-ant-2`.

Reply:
good catch, it was hardcoded everywhere. added a fix in latest commit.